clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -307,6 +285,36 @@ public HubConnectionState getConnectionState() {
         return stop(null);
     }
 
+    private void stopConnection(String errorMessage) {
+        HubException hubException = null;
+        hubConnectionStateLock.lock();
+        try {
+            hubConnectionState = HubConnectionState.DISCONNECTED;
```

Comment:
move this to just above the `finally`

Reply:
This should happen earlier in the stopping logic rather than later

Reply:
We should only set it once the connection is actually allowed to start again which would be right above the finally.